docs/user/vsphere/install_upi.md

```diff
@@ -0,0 +1,369 @@
+INFO API v1.12.4+c53f462 up
+INFO Waiting up to 30m0s for the bootstrap-complete event...
+```
+
+### Monitor for install completion
+
+The administrators can use the `wait-for install-complete` command of the OpenShift Installer to monitor install completion. The command succeeds when it notices that Cluster Version Operator has completed rolling out the OpenShift cluster from Kubernetes APIServer.
+
+```console
+$ openshift-install --dir test-vsphere wait-for install-complete
+INFO Waiting up to 30m0s for the cluster to initialize...
+```
+
+## Example vSphere UPI deployment
+
+Terraform [templates][upi-vsphere] provides an example of using OpenShift Installer to create a vSphere UPI OpenShift cluster.
+
+### Overview
+
+* Compute:
+    Uses `public` IPv4 addresses for each machine, so that all the machines are accessible on the Internet.
+
+* DNS and Load Balancing
+    Uses AWS [Route53](aws-route53) to configure the all the DNS records.
+    Uses Round-Robin DNS [RRDNS][rrdns] in place of load balancing solutions.
+
+Refer to the pre-requisites for using the example [here][upi-vsphere-example-pre-req]
+
+### Creating the cluster
+
+#### Installer assets
+
+Use the OpenShift Installer to create [Ignition configs][#getting-ignition-configs-for-machines] that will be used to create bootstrap, control plane and worker machines.
+
+#### Terraform variable file
+
+Use the [example][upi-vsphere-example-tfvar] `create_tfvars.sh` script to create a Terraform variable file, and edit the `tfvars` file on your favorite editor.
+
+```sh
+cd test-vsphere
+create_tfvars.sh
+```
+
+At a minimum, you will need to provide values for the following variables.
+* bootstrap_ignition_url
+* bootstrap_ip
+* control_plane_ips
+* compute_ips
+
+Move the `tfvars` file to the directory where the example terraform is.
+
+#### Creating resources
+
+Initialize terraform to download all the required providers. For more info on terraform [init][terraform-init] and terraform [providers][terraform-providers]
+
+```sh
+terraform init
+```
+
+Create all the resources using terraform by invoking [apply][terraform-apply]
+
+```sh
+terraform apply -auto-approve
+```
+
+#### Monitoring bootstrap-complete and removing bootstrap resources
+
+Use the bootstrap [monitoring][#monitor-for-bootstrap-complete] to track when cluster bootstrapping has finished. After the Kubernetes APIServer has been bootstrapped on the control plane machines, the bootstrap VM can be destroyed by following:
+
+```sh
+terraform apply -auto-approve -var 'boostrap_complete=true'
+```
+
+### Approving server certificates for nodes
+
+To allow Kube APIServer to communicate with the kubelet running on nodes for logs, rsh etc. The administrator needs to approve the CSR [requests][csr-requests] generated by each kubelet.
+
+You can approve all `Pending` CSR requests using,
+
+```sh
+oc get csr -ojson | jq -r '.items[] | select(.status == {} ) | .metadata.name' | xargs oc adm certificate approve
+```
+
```

Comment:
@staebler do we also need the [Updating image-registry to emptyDir storage backend](https://github.com/openshift/installer/blob/master/docs/user/metal/install_upi.md#updating-image-registry-to-emptydir-storage-backend) section added here? 
In my 4.1 on vSphere cluster, image-registry operator is not available and shows `Unable to apply resources: storage backend not configured` error. Or we're planning to use the vSphere volume directly later.

Reply:
The image-registry operator will be available. The image-registry will be able to create a PVC directly in vSphere.

Reply:
I guess that I was wrong about this. The image-registry needs a PVC with ReadWriteMany access, which is not supported by the vSphere cloud provider. I will add the section about setting the storage backend to `emptyDir`.

Reply:
Added with facfc5393...0ec07d076